File: scripts/run-migration.js
const { Client } = require('pg');
const fs = require('fs');
const path = require('path');

async function runMigration() {
  // Use DATABASE_URL from environment
  const dbUrl = process.env.DATABASE_URL;
  
  if (!dbUrl) {
    console.log('❌ DATABASE_URL not set');
    process.exit(1);
  }

  console.log('🔌 Connecting to Supabase...');
  
  const client = new Client({
    connectionString: dbUrl,
    ssl: { rejectUnauthorized: false }
  });

  try {
    await client.connect();
    console.log('✅ Connected!\n');

    // Read the migration SQL
    const migrationPath = path.join(__dirname, '..', 'supabase', 'migrations', '006_optimized.sql');
    const sql = fs.readFileSync(migrationPath, 'utf8');
    
    console.log('📄 Running migration 006_optimized.sql...\n');
    
    await client.query(sql);
    
    console.log('✅ Migration complete!\n');
    
    // Verify tables exist
    const tablesQuery = `
      SELECT table_name 
      FROM information_schema.tables 
      WHERE table_schema = 'public' 
      AND table_name IN ('organizations', 'organization_memberships', 'events', 'event_team_participants', 
                         'player_settings', 'recruiting_interests', 'conversations', 
                         'conversation_participants', 'messages')
      ORDER BY table_name;
    `;
    
    const tables = await client.query(tablesQuery);
    console.log('📊 Tables verified:');
    tables.rows.forEach(row => console.log(`   ✅ ${row.table_name}`));
    
  } catch (error) {
    console.error('❌ Error:', error.message);
  } finally {
    await client.end();
    console.log('\n🔌 Disconnected');
  }
}

runMigration();

File: scripts/run-migration.py
#!/usr/bin/env python3
import psycopg2
import os

# Connection details
configs = [
    {
        'name': 'Direct (IPv6)',
        'host': 'db.blspsttgyxuoqhskpmrg.supabase.co',
        'port': 5432,
        'user': 'postgres',
        'password': 'DrSgD9tC6D1ZilAZ',
        'dbname': 'postgres',
        'sslmode': 'require'
    },
    {
        'name': 'Session Pooler',
        'host': 'aws-0-us-east-1.pooler.supabase.com',
        'port': 5432,
        'user': 'postgres.blspsttgyxuoqhskpmrg',
        'password': 'DrSgD9tC6D1ZilAZ',
        'dbname': 'postgres',
        'sslmode': 'require'
    }
]

# Read migration SQL
script_dir = os.path.dirname(os.path.abspath(__file__))
migration_path = os.path.join(script_dir, '..', 'supabase', 'migrations', '006_optimized.sql')
with open(migration_path, 'r') as f:
    sql = f.read()

for config in configs:
    print(f"\n🔌 Trying {config['name']}...")
    try:
        conn = psycopg2.connect(
            host=config['host'],
            port=config['port'],
            user=config['user'],
            password=config['password'],
            dbname=config['dbname'],
            sslmode=config['sslmode'],
            connect_timeout=30
        )
        conn.autocommit = True
        print("✅ Connected!")
        
        cur = conn.cursor()
        print("📄 Running migration...")
        cur.execute(sql)
        print("✅ Migration complete!")
        
        # Verify tables
        cur.execute("""
            SELECT table_name FROM information_schema.tables 
            WHERE table_schema = 'public' 
            AND table_name IN ('organizations', 'events', 'player_settings', 'conversations', 'messages')
            ORDER BY table_name;
        """)
        
        print("\n📊 Tables verified:")
        for row in cur.fetchall():
            print(f"   ✅ {row[0]}")
        
        cur.close()
        conn.close()
        print("\n🎉 Success!")
        exit(0)
        
    except Exception as e:
        print(f"   ❌ {str(e)[:100]}")

print("\n⚠️ All connection methods failed")
exit(1)

